application_sdk/activities/metadata_extraction/base.py

```diff
@@ -0,0 +1,102 @@
+from typing import Any, Dict, Optional, Type
+
+from temporalio import activity
+
+from application_sdk.activities import ActivitiesInterface, ActivitiesState
+from application_sdk.activities.common.utils import get_workflow_id
+from application_sdk.clients.base import BaseClient
+from application_sdk.common.credential_utils import get_credentials
+from application_sdk.constants import APP_TENANT_ID, APPLICATION_NAME
+from application_sdk.handlers.base import BaseHandler
+from application_sdk.observability.logger_adaptor import get_logger
+from application_sdk.transformers import TransformerInterface
+
+logger = get_logger(__name__)
+activity.logger = logger
+
+
+class BaseMetadataExtractionActivitiesState(ActivitiesState):
+    """State for base metadata extraction activities."""
+
+    client: Optional[Type[BaseClient]] = None
+    handler: Optional[Type[BaseHandler]] = None
+    transformer: Optional[Type[TransformerInterface]] = None
```

Comment:
these are typed by classes by you're assigning them classes instances [here](https://github.com/atlanhq/application-sdk/pull/660/files#diff-65c034e828fa96f2e5a0a1e7ba020a82397df9c08e75eab99b3a64e421885ad8R89-R102)